addons/bus/static/src/workers/election_worker.js

```diff
@@ -0,0 +1,120 @@
+import { Deferred } from "@bus/workers/bus_worker_utils";
+
+export class ElectionWorker {
+    MAIN_TAB_TIMEOUT_PERIOD = 3000;
+
+    constructor() {
+        this.masterTab = null;
+        this.candidates = new Set();
+        this.lastHeartbeat = Date.now();
+        this.electionDeferred = new Deferred();
+        this.masterReplyDeferred = null;
+        this.heartbeatCheckInterval = null;
+        this.heartbeatRequestInterval = null;
+    }
```

Comment:
Nitpick: can define params in scope of class rather than in `constructor()`:

```js
masterTab = null;
candidates = new Set();
...
```
Having code in `constructor()` is useful when ordering matters or want to write complex code that just plain attribute initialization.